src/reference/informative/informC02_units_reference.rst

```diff
@@ -8,4 +8,300 @@
+            <variables name="water" units="spoonful" />
+            <variables name="salt" units="dash" />
+            <variables name="lavender_flowers" units="smidgen" />
+            ...
+          </component>
+
+          <component name="CheeseAndAlmondCrust">
+            ...
+          </component>
+
+          <!-- Local units definitions for spoonful, dash, and smidgen. -->
+          <units name="spoonful">
+            <unit units="litre" prefix="milli" multiplier="15" />
+          </units>
+          <units name="dash">
+            <unit units="gram" multiplier="5" />
+          </units>
+          <units name="smidgen">
+            <unit units="gram" multiplier="1" />
+          </units>
+          <units name="dollop">
+            <unit units="litre" prefix="milli" multiplier="20" />
+          </units>
+        </model>
+
+        <!-- Inside the file "filling_recipes.cellml": -->
+        <model name="PieFillingRecipes">
+
+          <component name="BlueberryCinnamonFilling">
+            <!-- These units are built-in, so do not change.  -->
+            <variables name="blueberries" units="gram" />
+            <variables name="sugar" units="dimensionless" />
+            <variables name="cornflour" units="gram" />
+
+            <!-- These units are defined for use in this, their local scope, below. -->
+            <variables name="cinnamon" units="smidgen" />
+            <variables name="water" units="spoonful" />
+
+            <math>
+                ...
+            </math>
+          </component>
+
+          <component name="AppleAndPearFilling">
+            ...
+          </component>
+
+          <!-- Local units definitions for spoonful and smidgen. -->
+          <units name="spoonful">
+            <unit units="litre" prefix="milli" multiplier="5" />
+          </units>
+          <units name="smidgen">
+            <unit units="gram" multiplier="20" />
+          </units>
+
+        </model>
+
+    This is where the idea of *scope* becomes important.  
+    As it stands, there is no conflict between the two different definitions of :code:`spoonful` and :code:`dash`, because each of the components refers to *its own definition* of these units.
+    The components do not "know" that there is any other definition out there, because they cannot "see" up into the importing model.
+
+    Now let's consider that the cook wants to alter the recipe a little after these two main ingredients have been imported, by adding a spoonful of brandy to some custard.
+    The top-level model becomes:
+
+    .. code::
+
+      model: BlueberryPieRecipe
+        ├─ component: BrandyCustard
+        │    ├─ variable: custard (litre)
+        │    └─ variable: brandy           
+        │         └─ units: spoonful  # These units are not defined in a scope
+        │                               which this component can access: 
+        │                               the model is invalid.
```

Comment:
Actually, `spoonful` is defined according to your previous diagram. You must have rearranged your recipes at some point... :)

Reply:
Oops, I take this back!